pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java

```diff
@@ -0,0 +1,317 @@
+      }
+
+      // Fall back to remote address
+      return remoteAddr;
+    } catch (Exception e) {
+      LOG.debug("Failed to extract client IP address", e);
+      return "unknown";
+    }
+  }
+
+  /**
+   * Extracts user ID from request headers.
+   * Looks for common authentication headers.
+   *
+   * @param requestContext the container request context
+   * @return the user ID or "anonymous" if not found
+   */
+  private String extractUserId(ContainerRequestContext requestContext) {
+    try {
+      // Check for common user identification headers
+      String authHeader = requestContext.getHeaderString("Authorization");
+      if (StringUtils.isNotBlank(authHeader)) {
+        // For basic auth, extract username; for bearer tokens, use a placeholder
+        if (authHeader.startsWith("Basic ")) {
+          // Could decode basic auth to get username, but for security keep it as placeholder
+          return "basic-auth-user";
+        } else if (authHeader.startsWith("Bearer ")) {
+          return "bearer-token-user";
+        }
+      }
+
+      // Check for custom user headers
+      String userHeader = requestContext.getHeaderString("X-User-ID");
+      if (StringUtils.isNotBlank(userHeader)) {
+        return userHeader.trim();
+      }
+
+      userHeader = requestContext.getHeaderString("X-Username");
+      if (StringUtils.isNotBlank(userHeader)) {
+        return userHeader.trim();
+      }
+
+      return ANONYMOUS;
+    } catch (Exception e) {
+      LOG.debug("Failed to extract user ID", e);
+      return ANONYMOUS;
+    }
+  }
+
+  /**
+   * Extracts service ID from request headers.
+   * Service ID should be provided by the client in headers, not from configuration.
+   *
+   * @param requestContext the container request context
+   * @return the service ID or "unknown" if not found
+   */
+  private String extractServiceId(ContainerRequestContext requestContext) {
+    try {
+      // Check for custom service ID headers
+      String serviceId = requestContext.getHeaderString("X-Service-ID");
+      if (StringUtils.isNotBlank(serviceId)) {
+        return serviceId.trim();
+      }
+
+      serviceId = requestContext.getHeaderString("X-Service-Name");
+      if (StringUtils.isNotBlank(serviceId)) {
+        return serviceId.trim();
+      }
+
+      return "unknown";
+    } catch (Exception e) {
+      LOG.debug("Failed to extract service ID", e);
+      return "unknown";
+    }
+  }
+
+  /**
+   * Captures the request payload for audit logging based on configuration.
+   * Uses dynamic configuration to control what data is captured.
+   *
+   * @param requestContext the request context
+   * @return the captured request payload
+   */
+  private Object captureRequestPayload(ContainerRequestContext requestContext) {
+    // Get current configuration (fallback to defaults if no config manager)
+    AuditConfig config = _configManager != null ? _configManager.getCurrentConfig() : new AuditConfig();
```

Comment:
do we expect `_configManager` to be null? if so, won't other functions in this class need null handling as well?